File: services/api-laravel/app/Http/Controllers/MetricsController.php
<?php

namespace App\Http\Controllers;

use App\Support\Metrics;
use Illuminate\Http\Response;

class MetricsController extends Controller
{
    // I-13: Prometheus scrape endpoint.
    public function index(): Response
    {
        return response(Metrics::render(), 200)
            ->header('Content-Type', 'text/plain; version=0.0.4');
    }
}

File: services/api-phoenix/lib/app_web/controllers/metrics_controller.ex
defmodule AppWeb.MetricsController do
  use Phoenix.Controller, formats: [:json]

  # I-13: Prometheus scrape endpoint. Exempt from auth and rate limiting.
  def index(conn, _params) do
    conn
    |> put_resp_content_type("text/plain")
    |> send_resp(200, AppWeb.Metrics.render())
  end
end
